sys/fido2/ctap/ctap.c

```diff
@@ -0,0 +1,1508 @@
+/*
+ * Copyright (C) 2021 Nils Ollrogge
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap
+ * @{
+ * @file
+ *
+ * @author  Nils Ollrogge <nils-ollrogge@outlook.de>
+ * @}
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+
+#include "fmt.h"
+#include "xtimer.h"
+#include "byteorder.h"
+
+#include "fido2/ctap/transport/ctap_transport.h"
+#include "fido2/ctap.h"
+#include "fido2/ctap/ctap_utils.h"
+#include "fido2/ctap/ctap_cbor.h"
+#include "fido2/ctap/ctap_mem.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/*** main CTAP2 functions ***/
```

Comment:
Yes all the main functions are defined in the spec. The helper functions are specific to this implementation. 
I think you are suggesting to define all methods defined in the CTAP spec publicly in `ctap.h`? This might be useful when the code is used for an internal authenticator that does not communicate over USB/NFC/BLE with the host. Otherwise only exposing `fido2_ctap_handle_request` should be enough, as the communication layers do not know which kind of CTAP method is being called.